x-pack/plugins/observability_solution/observability_ai_assistant_app/public/hooks/use_last_used_prompts.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { uniq } from 'lodash';
+import { useKibana } from './use_kibana';
+
+const AI_ASSISTANT_LAST_USED_PROMPT_STORAGE = 'ai-assistant-last-used-prompts';
+
+export function useLastUsedPrompts() {
+  const { storage } = useKibana().services;
+
+  const previousPrompts = (storage.get(AI_ASSISTANT_LAST_USED_PROMPT_STORAGE) as string[]) ?? [];
+
+  return {
+    previousPrompts,
+    addLastUsed: (prompt: string) => {
```

Comment:
should this be in a `useCallback`, so you can more easily pass it as a dependency without invalidating those dependencies on every render?

Reply:
If I do this `lastUsedPrompts` needs to be wrapped in a useMemo. Wonder if it's worth it.